include/cpuinfo_x86.h

```diff
@@ -81,12 +81,44 @@ typedef struct {
   // Make sure to update X86FeaturesEnum below if you add a field here.
 } X86Features;
 
+// Increase this value if more cache levels are needed.
+#ifndef CPU_FEATURES_MAX_CACHE_LEVEL
+#define CPU_FEATURES_MAX_CACHE_LEVEL 10
+#endif
+
+typedef enum {
+  EMPTY = 0,
+  DATA_CACHE = 1,
+  INSTRUCTION_CACHE = 2,
+  UNIFIED_CACHE = 3,
+  TLB = 4,
+  DTLB = 5,
+  STLB = 6,
+  PREFETCH = 7
+} X86CacheType;
+
+typedef struct {
+  int level;
+  X86CacheType cache_type;
+  int cache_size;  // Cache size in kBytes
+  int ways;        // Associativity, 0 undefined, 0xFF fully associative
+  int line_size;   // Cache line size in bytes
+  int entries;     // number of entries for TLB
+  int partitioning;
+} X86CacheLevelInfo;
+
+typedef struct {
+  int size;
+  X86CacheLevelInfo levels[CPU_FEATURES_MAX_CACHE_LEVEL];
+} X86CacheInfo;
```

Comment:
I'm tempted to move these structs to a separate header so we can enumerate cache info for all architectures.
Then I would offer a separate `GetX86CacheInfo` function so you don't pay for it if you don't need it.